op-acceptor/runner/runner.go

```diff
@@ -688,6 +641,108 @@ func (r *runner) parseTestOutput(output []byte, metadata types.ValidatorMetadata
 	return result
 }
 
+// parseTestEvent parses a single line of test output into a TestEvent
+func parseTestEvent(line []byte) (TestEvent, error) {
+	var event TestEvent
+	err := json.Unmarshal(line, &event)
+	return event, err
+}
+
+// isMainTestEvent checks if the event belongs to the main test or package
+func isMainTestEvent(event TestEvent, mainTestName string) bool {
+	return event.Test == "" || event.Test == mainTestName
+}
+
+// processMainTestEvent handles events for the main test
+func processMainTestEvent(event TestEvent, result *types.TestResult, testStart, testEnd *time.Time,
+	errorMsg *strings.Builder, hasSkip *bool) {
+	switch event.Action {
+	case "start":
+		*testStart = event.Time
+	case "pass":
+		*testEnd = event.Time
+		result.Status = types.TestStatusPass
+	case "fail":
+		*testEnd = event.Time
+		result.Status = types.TestStatusFail
+	case "skip":
+		*testEnd = event.Time
+		result.Status = types.TestStatusSkip
+		*hasSkip = true
+	case "output":
+		if event.Output != "" {
+			errorMsg.WriteString(event.Output)
+		}
+	}
+}
+
+// processSubTestEvent handles events for subtests
+func processSubTestEvent(event TestEvent, result *types.TestResult,
+	subTestStatuses map[string]types.TestStatus, hasSkip *bool) {
+	subTest, exists := result.SubTests[event.Test]
+	if !exists {
+		subTest = &types.TestResult{
+			Metadata: types.ValidatorMetadata{
+				FuncName: event.Test,
+				Package:  result.Metadata.Package,
+			},
+			Status: types.TestStatusPass, // Default to pass
+		}
+		result.SubTests[event.Test] = subTest
+	}
+
+	switch event.Action {
```

Comment:
I don't recall the shape of the output, but just to confirm: there's no "start" for subtests ?

Reply:
Subtests have a start (all JSON events start with start, I believe), but it just had timing information.
Added in https://github.com/ethereum-optimism/infra/pull/257